app/controllers/admin/projects_controller.rb

```diff
@@ -95,13 +90,4 @@ def sort
     end
     render partial: 'projects'
   end
-
-  private
-
-  def redirect_if_not_owner_or_admin(project)
-    unless current_user.owner?(project) || current_user.administrator?
-      flash[:error] = "Project details can only be altered by the owner (#{project.user.login}) or an administrator"
-      redirect_to project_path(project)
-    end
-  end
 end
```

Comment:
I'm not handling all redirection centrally. It means the error messages change slightly, but avoids littering all out controllers with sections like this.